### ISSUE TITLE
View the output of two simultaneous instances side-by-side

### ISSUE THREAD

**Issue by Glenugie** [thennothinghappened/GMEdit-Constructor#21]
If doing testing of netcode, which requires two instances of the game to be run, the current workflow involves starting one, then tabbing back to the editor to start another once compile of the first is finished (quickly starting two results in errors, I assume concurrent read/write of files). Ideally if there was a configuration option to automatically run 2 (or X) instances on press of a hotkey, that would save time.

Additionally, right now I have to run without reusing the output window (to see the output of both instances), which means manually closing the output windows to not clog the editor. As part of the same feature, this could be resolved by having the outputs of the concurrent instances merged into one and qualifying each line with some identifier of which instance it came from.

**Comment by thennothinghappened** [comment]
Hello!

As of [v0.17.0](https://github.com/thennothinghappened/GMEdit-Constructor/releases/tag/v0.17.0), if you disable the preference to re-use tabs, you can hit F5 repeatedly and it will spawn multiple `Run` jobs. This isn't as ideal as I'd like and could stand to have better ergonomics, but hopefully this tides things over and is a lot less cumbersome than waiting for a job to finish compilation before starting a companion one.

**Comment by Glenugie** [comment]
Appreciated, it will certainly make quick testing a bit less arduous. In the interim I had put together an AutoHotKey script (bound to F6) that would run two instances and place them side-by-side for quicker testing. Posting it here in case it's of use to anyone finding this issue who has a similar set up:
```
#If WinActive("ahk_exe GMEdit.exe") 
	F6::
		Send {F5}
		loops := 0
		; Wait for the game to be active (meaning it's compiled)
		while (loops < 100) {
			WinGet, winPName, ProcessName, A
			if (winPName == "Runner.exe") {
				break
			}
			loops += 1
			Sleep, 100
		}

		; Sent it to the left
		Sleep, 500
		Send {LWin down}
		Send {Left}
		Send {LWin up}
		Sleep, 500
		Send {Escape}
		Sleep, 500

		; Alt tab back to GMEdit
		Send {Alt down}
		Send {Tab}
		Send {Alt up}

		; Wait for GMEdit to be active
		loops = 0
		while (loops < 100) {
			WinGet, winPName, ProcessName, A
			if (winPName == "GMEdit.exe") {
				break
			}
			loops += 1
			Sleep, 100
		}
		Sleep, 500

		Send {F5}
		loops = 0
		; Wait for the game to be active (meaning it's compiled)
		while (loops < 100) {
			WinGet, winPName, ProcessName, A
			if (winPName == "Runner.exe") {
				break
			}
			loops += 1
			Sleep, 100
		}
		; Send it to the right
		Sleep, 500
		Send {LWin down}
		Send {Right}
		Send {LWin up}
		Sleep, 500
		Send {Escape}
		Sleep, 500

		; Alt tab back to the first game window
		Send {Alt down}
		Send {Tab}
		Sleep, 100
		Send {Tab}
		Send {Alt up}
	return
#If
```

**Comment by thennothinghappened** [comment]
As of under a week ago, GMEdit on `master` now has basic support for having multiple windows. If this extends further to getting multiple windows which can support multiple tabs, and have hotkeys work, this would be the preferred solution in my books for wanting to see two outputs simultaneously. This would likely require some work on my side to make the plugin compatible with launching from an auxiliary window (mainly to do with singletons), so I'll keep this issue open for that, or if any other ideas come up.
